compiler/rustc_passes/src/check_attr.rs

```diff
@@ -2517,11 +2513,6 @@ fn check_invalid_crate_level_attr(tcx: TyCtxt<'_>, attrs: &[Attribute]) {
         sym::start,
         sym::rustc_main,
         sym::unix_sigpipe,
```

Comment:
What is this remaining list?
Most built-in attributes are invalid in crate root position, but only a few are listed here.

Reply:
Any good idea for this?
I have found we have 
https://github.com/rust-lang/rust/blob/04521fd10e6eefb113ca1cefd02862599fd08640/compiler/rustc_feature/src/builtin_attrs.rs#L271

I tried to add a new helper for query:

```rust
pub fn is_normal_builtin_attr(name: Symbol) -> bool {
    BUILTIN_ATTRIBUTE_MAP.get(&name).is_some_and(|attr| match attr.type_ {
        AttributeType::CrateLevel => false,
        AttributeType::Normal => true,
    })
}
```

It turns out `Normal` in `BUILTIN_ATTRIBUTES` means the attribute may used at `CrateLevel` or other item level, for example `warn`, `allow`, etc.

`CrateLevel` is right(means it can only be used at crate level).

We need a list of builtin attributes which can not be used at crate level...

Reply:
Ok, let's keep this list as is for now.